src/utils.py

```diff
@@ -29,28 +37,71 @@ def format_token_short(token):
 
 
 def format_integer(number):
-  return str(number) # TODO: Format better the numbers
+  return '{:,d}'.format(number)
+
+def format_amount_in_weis(amount, decimals, rounding=ROUND_DOWN):
+  value = amount / Decimal(10 ** decimals)
+  return format_amount(value, decimals=decimals, rounding=rounding)
 
-def format_amount_in_weis(amount, decimals):
-  return str(amount / (10 ** decimals)) # TODO: Format better the weis
+def format_price(amount, decimals=10, rounding=ROUND_DOWN, currency=''):
+  price = format_amount(amount, decimals=decimals, rounding=rounding)
+  return price + ' ' + currency if currency else price
+
+def format_amount(amount, decimals=18, rounding=ROUND_DOWN):
+  quantize_value = Decimal(10) ** -Decimal(decimals)
+  rounded_value = Decimal(amount).quantize(quantize_value, rounding=rounding)
+
+  return f'{{:,.{decimals}f}}'.format(rounded_value).rstrip('0').rstrip('.')
```

Comment:
Yikes, f string and format?

Reply:
Didn't know about the formatting. I'll address in https://github.com/gnosis/dex-cli/issues/39